src/MutationExpressionHandler.js

```diff
@@ -0,0 +1,31 @@
+import PathProxy from './PathProxy';
+import MutationExpressionCallbackHandler from './MutationExpressionCallbackHandler';
+
+/**
+ * Returns a function that creates a mutation expression
+ * with the current path expression as domain
+ * and the given arguments as range.
+ * These arguments can either be raw, or other path expressions.
+ */
+export default class MutationExpressionHandler {
+  constructor(mutationType) {
```

Comment:
I'd personally solve this with subclassing. Have one `InsertExpressionHandler` and one `DeleteExpressionHandler` which inherit from `MutationExpressionHandler` and have a property `mutationType`?